File: notification-service/src/main/java/com/eirsteir/coffeewithme/notification/service/CoffeeBreakEventConsumer.java
package com.eirsteir.coffeewithme.notification.service;

import com.eirsteir.coffeewithme.commons.domain.coffeebreak.CoffeeBreakCreatedEvent;
import com.eirsteir.coffeewithme.commons.domain.user.UserDetails;
import io.eventuate.tram.events.subscriber.DomainEventEnvelope;
import io.eventuate.tram.events.subscriber.DomainEventHandlers;
import io.eventuate.tram.events.subscriber.DomainEventHandlersBuilder;

public class CoffeeBreakEventConsumer extends EventConsumer {

  public DomainEventHandlers domainEventHandlers() {
    return DomainEventHandlersBuilder.forAggregateType(
            "com.eirsteir.coffeewithme.social.domain.coffeebreak.CoffeeBreak")
        .onEvent(CoffeeBreakCreatedEvent.class, this::handleCoffeeBreakCreatedEvent)
        .build();
  }

  private void handleCoffeeBreakCreatedEvent(
      DomainEventEnvelope<CoffeeBreakCreatedEvent> domainEventEnvelope) {
    CoffeeBreakCreatedEvent coffeeBreakCreatedEvent = domainEventEnvelope.getEvent();
    for (UserDetails userDetails : coffeeBreakCreatedEvent.getCoffeeBreakDetails().getAddressees())
      super.handleEvent(coffeeBreakCreatedEvent);
  }
}

File: notification-service/src/main/kotlin/com/eirsteir/coffeewithme/notification/service/CoffeeBreakEventConsumer.kt
package com.eirsteir.coffeewithme.notification.service

import com.eirsteir.coffeewithme.commons.domain.coffeebreak.CoffeeBreakCreatedEvent
import com.eirsteir.coffeewithme.notification.repository.NotificationRepository
import io.eventuate.tram.events.subscriber.DomainEventEnvelope
import io.eventuate.tram.events.subscriber.DomainEventHandlers
import io.eventuate.tram.events.subscriber.DomainEventHandlersBuilder
import org.modelmapper.ModelMapper
import org.springframework.messaging.simp.SimpMessagingTemplate

class CoffeeBreakEventConsumer(notificationRepository: NotificationRepository, template: SimpMessagingTemplate,
                               modelMapper: ModelMapper
)
    : EventConsumer(notificationRepository, template, modelMapper) {

    fun domainEventHandlers(): DomainEventHandlers {
        return DomainEventHandlersBuilder.forAggregateType(
            "com.eirsteir.coffeewithme.social.domain.coffeebreak.CoffeeBreak"
        )
            .onEvent(CoffeeBreakCreatedEvent::class.java) { domainEventEnvelope: DomainEventEnvelope<CoffeeBreakCreatedEvent?>? ->
                handleCoffeeBreakCreatedEvent(
                    domainEventEnvelope
                )
            }
            .build()
    }

    private fun handleCoffeeBreakCreatedEvent(
        domainEventEnvelope: DomainEventEnvelope<CoffeeBreakCreatedEvent?>?
    ) {
        val coffeeBreakCreatedEvent = domainEventEnvelope?.event
        for (user in coffeeBreakCreatedEvent?.coffeeBreakDetails?.addressees!!)
            super.handleEvent(
            coffeeBreakCreatedEvent
        )
    }
}